File: d01/prompt-engineering/python/01-basic-interactions/02-prompt-with-ai-role/02-prompt-with-ai-role.py
#!/usr/bin/env python3
"""
Example: single-prompt chat completion
Run with:
    export OPENAI_API_KEY="sk-..."   # or set it in your shell profile
    python 02-prompt-with-ai-role.py
"""

import os
from openai import OpenAI

# 1️⃣ Get the OpenAI API key from the environment (raises if not set)
api_key = os.environ["OPENAI_API_KEY"]

# 2️⃣ Create a client instance
client = OpenAI(api_key=api_key)

# 3️⃣ Define the user prompt
prompt = "What are the core AI components and their relations?"

# 4️⃣ Call the chat completion endpoint
response = client.chat.completions.create(
    model="gpt-4.1",
    messages=[
          {"role": "system", "content": "Act as an experienced AI trainer"},
          {"role": "user", "content": prompt}
    ]
)

# 5️⃣ Display the assistant’s answer
print(response.choices[0].message.content)

File: d01/prompt-engineering/typescript/01-basic-interactions/02-prompt-with-ai-role/02-prompt-with-ai-role.ts
import { OpenAI } from "openai";
import dotenv from 'dotenv';
import * as process from "node:process";

async function main() {
    dotenv.config({path: '../../.env'});

    // gets the API key from the OPENAI_API_KEY environment variable
    const openai = new OpenAI();

    const usedModel = process.env["USED_MODEL"] ?? 'gpt-4.1';
    const systemInstructions = "Act as an experienced AI trainer, describe in a high-level overview";
    const prompt = "What are the relationships between AI, ML, DL and NLP?"

    console.log(`Querying '${usedModel}' for '${prompt}', using the system instructions '${systemInstructions}'...`);
    const response = await openai.chat.completions.create({
        model: usedModel,
        messages: [
            {"role": "system", "content": systemInstructions},
            {"role": "user", "content": prompt}
        ]
    });

    console.log(response.choices[0]?.message.content);
}

main().catch(error => {
    console.error(error.error.message);
});